packages/core/src/utils.ts

```diff
@@ -16,115 +15,6 @@ import type { CoreTool, ToolAction, VercelTool } from './tools';
 
 export const delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
 
-export function jsonSchemaPropertiesToTSTypes(value: any): z.ZodTypeAny {
```

Comment:
Looks like this was added for composio https://github.com/mastra-ai/mastra/commit/70f88eae#diff-6552eacdc0f4378c00e24a96877df8352eeb0db006852dfdcf511f59d9b781aeR5
And later re-used for MCP tools. Now we don't have a composio integration so it makes sense to remove this and use `json-schema-to-zod` instead, like we do elsewhere.